File: server/ruby/app/infrastructure/mappers/payment_details/payment_product5402_specific_output_mapper.rb
module Infrastructure
  module Mappers
    module PaymentDetails
      class PaymentProduct5402SpecificOutputMapper
        def self.from_sdk_response(response)
          return nil if response.nil?

          Business::Domain::Payments::PaymentDetails::PaymentProduct5402SpecificOutput.new(
            brand: response.brand
          )
        end
      end
    end
  end
end

File: server/python/app/infrastructure/mappers/payment_details/payment_product_5402_specific_output_mapper.py
from typing import Optional

from app.application.domain.payments.payment_details.payment_product_5402_specific_output import PaymentProduct5402SpecificOutput
from onlinepayments.sdk.domain.payment_product5402_specific_output import PaymentProduct5402SpecificOutput as PaymentProduct5402SpecificOutputSdk

class PaymentProduct5402SpecificOutputMapper:

    @staticmethod
    def map_from_sdk_response(
        response: Optional[PaymentProduct5402SpecificOutputSdk],
    ) -> Optional[PaymentProduct5402SpecificOutput]:
        if response is None:
            return None

        return PaymentProduct5402SpecificOutput(
            brand = response.brand
        )